shared-libs/lineage/test/integration.js

```diff
@@ -795,6 +795,18 @@ describe('Lineage', function() {
         });
     });
 
+    it('should not recurse more than needed', () => {
+      const copy = (doc) => JSON.parse(JSON.stringify(doc));
+      return Promise
+        .all([
+          lineage.hydrateDocs([circular_report]),
+          lineage.hydrateDocs([copy(circular_report), copy(circular_report)])
+        ])
+        .then(results => {
+          expect(results[0][0]).to.deep.equal(results[1][0]);
```

Comment:
should we also assert `expect(results[0][0]).to.deep.equal(results[1][1]);`?

Reply:
I added your suggested assertion.